### ISSUE TITLE
Updating an asset inside an image property not possible

### ISSUE THREAD

**Issue by pduersteler** [neos/neos-development-collection#2295]
### Description

When clicking the image preview of a node's image property in the inspector, the asset's edit view is shown (with title, description, tags, collections, …). Changing the title or the description in this view does nothing.

### Steps to Reproduce

1. Select a node that has an image property
2. Assign an image if none is assigned already
3. Click on the preview image of said property inside the inspector to open the "edit view"
4. Change title and/or description of the asset
5. Press "save"

#### Expected behavior

Changed fields are updated in the database.

#### Actual behavior

Changed fields are not updated; If one reopens the "edit view" or edits the asset through the media library, old values are displayed.

### Affected Versions

neos/neos 4.1.8
neos/media 4.1.8
neos/media-browser 4.1.8
neos/ui dev-master@35f0471
neos/ui-compiled dev-master@46f0868

**Comment by kdambekalns** [comment]
The problem becomes clear when looking at the log: summoning the edit dialog that way opens it for an ImageVariant, and that cannot have it's title set, and thus runs into

```
Exception #1297759968 in line 210 of /path/to/site/Data/Temporary/Production/Cache/Code/Flow_Object_Classes/Neos_Flow_Mvc_Controller_Argument.php: Could not convert target type "Neos\Media\Domain\Model\Asset": Setting the title on an ImageVariant is not supported.

11 Neos\Flow\Property\PropertyMapper_Original::convert(array|4|, "Neos\Media\Domain\Model\Asset", Neos\Flow\Mvc\Controller\MvcPropertyMappingConfiguration)
10 Neos\Flow\Mvc\Controller\Argument_Original::setValue(array|4|)
9 Neos\Flow\Mvc\Controller\AbstractController_Original::mapRequestArgumentsToControllerArguments()
8 Neos\Flow\Mvc\Controller\ActionController_Original::processRequest(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
7 Neos\Flow\Mvc\Dispatcher_Original::initiateDispatchLoop(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
6 Neos\Flow\Mvc\Dispatcher_Original::dispatch(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
5 Neos\Flow\Mvc\DispatchComponent_Original::handle(Neos\Flow\Http\Component\ComponentContext)
4 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
3 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
2 Neos\Flow\Http\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()

Previous exception: Exception #1366627475 in line 259 of /path/to/site/Packages/Libraries/neos/utility-objecthandling/Classes/ObjectAccess.php: Setting the title on an ImageVariant is not supported.

16 Neos\Media\Domain\Model\ImageVariant_Original::setTitle("some title")
15 Neos\Utility\ObjectAccess::setProperty(Neos\Media\Domain\Model\ImageVariant, "title", "some title")
14 Neos\Flow\Property\TypeConverter\PersistentObjectConverter_Original::convertFrom(array|4|, "Neos\Media\Domain\Model\Asset", array|3|, Neos\Flow\Mvc\Controller\MvcPropertyMappingConfiguration)
13 Neos\Media\TypeConverter\AssetInterfaceConverter_Original::convertFrom(array|4|, "Neos\Media\Domain\Model\Asset", array|3|, Neos\Flow\Mvc\Controller\MvcPropertyMappingConfiguration)
12 Neos\Flow\Property\PropertyMapper_Original::doMapping(array|4|, "Neos\Media\Domain\Model\Asset", Neos\Flow\Mvc\Controller\MvcPropertyMappingConfiguration, array|0|)
11 Neos\Flow\Property\PropertyMapper_Original::convert(array|4|, "Neos\Media\Domain\Model\Asset", Neos\Flow\Mvc\Controller\MvcPropertyMappingConfiguration)
10 Neos\Flow\Mvc\Controller\Argument_Original::setValue(array|4|)
9 Neos\Flow\Mvc\Controller\AbstractController_Original::mapRequestArgumentsToControllerArguments()
8 Neos\Flow\Mvc\Controller\ActionController_Original::processRequest(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
7 Neos\Flow\Mvc\Dispatcher_Original::initiateDispatchLoop(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
6 Neos\Flow\Mvc\Dispatcher_Original::dispatch(Neos\Flow\Mvc\ActionRequest, Neos\Flow\Http\Response)
5 Neos\Flow\Mvc\DispatchComponent_Original::handle(Neos\Flow\Http\Component\ComponentContext)
4 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
3 Neos\Flow\Http\Component\ComponentChain_Original::handle(Neos\Flow\Http\Component\ComponentContext)
2 Neos\Flow\Http\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()


HTTP REQUEST:
POST /neos/media/browser/images/update.html HTTP/1.1
Host: www.site.com
Upgrade-Insecure-Requests: 1
Connection: keep-alive
Content-Length: 896
Content-Type: application/x-www-form-urlencoded
Referer: https://www.site.com/neos/media/browser/images/edit.html?asset[__identity]=54d8de2c-4dd5-46cf-9a8a-f9c473bdc184
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.7,de;q=0.3
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:66.0) Gecko/20100101 Firefox/66.0

asset%5B__identity%5D=54d8de2c-4dd5-46cf-9a8a-f9c473bdc184&__referrer%5B%40package%5D=Neos.Media.Browser&__referrer%5B%40subpackage%5D=&__referrer%5B%40controller%5D=Image&__referrer%5B%40action%5D=edit&__referrer%5Barguments%5D=YToxOntzOjU6ImFzc2V0IjthOjE6e3M6MTA6Il9faWRlbnRpdHkiO3M6MzY6IjU0ZDhkZTJjLTRkZDUtNDZjZi05YThhLWY5YzQ3M2JkYzE4NCI7fX0%3D11ac21294974ffdd977368d55f9b1d7c02af00df&__trustedProperties=a%3A2%3A%7Bs%3A5%3A%22asset%22%3Ba%3A4%3A%7Bs%3A5%3A%22title%22%3Bi%3A1%3Bs%3A7%3A%22caption%22%3Bi%3A1%3Bs%3A15%3A%22copyrightNotice%22%3Bi%3A1%3Bs%3A10%3A%22__identity%22%3Bi%3A1%3B%7Di%3A0%3Bi%3A1%3B%7Daf270af4e9442593579da5b2a5805af42ab35295&__csrfToken=860e076ffed533f484c73b73616dcbcf&asset%5Btitle%5D=some+title&asset%5Bcaption%5D=some+caption&asset%5BcopyrightNotice%5D=some+copyright

HTTP RESPONSE:
[response was empty]

PHP PROCESS:
Inode: 11931
PID: 53308
UID: 50001
GID: 50001
User: ci

    [array] =>
        previousException:
        [string] => Exception #1366627475 in line 259 of /path/to/site/Packages/Libraries/neos/utility-objecthandling/Classes/ObjectAccess.php: Setting the title on an ImageVariant is not supported.
```

**Comment by kdambekalns** [comment]
Questions:

- was it always the case, that clicking on the thumbnail in the inspector opened the edit view?
- would it even be desired to be able to edit the asset behind the variant? It could change places the editor does not expect to edit at all…

**Comment by kdambekalns** [comment]
So, it turns out this is actually fixed with https://github.com/neos/neos-ui/pull/2340, that's why in my tests with the Neos demo it worked even with 4.2 (I used `dev-master` of the Neos UI).

**Comment by kdambekalns** [comment]
Fixed as of 3.2.1 and 2.2.0, respectively.
